frontend/tests/controllers/BadgesTest.php

```diff
@@ -27,16 +27,18 @@ private static function cleanDb() {
     }
 
     private static function RunRequests($apicall) {
-        $identity = new stdClass();
-        $identity->username = $apicall['username'];
-        $identity->password = $apicall['password'];
-        $login = self::login($identity);
+        $login = self::login(new Identities([
+            'username' => $apicall['username'],
+            'password' => $apicall['password'],
+        ]));
         foreach ($apicall['requests'] as $req) {
-            $r = new Request();
+            $params['auth_token'] = $login->auth_token;
```

Comment:
no falta declarar `$params`?

```suggestion
            $params = [
                'auth_token' => $login->auth_token,
            ];
```

Reply:
Cierto!
Done